resources/js/web3_init.js

```diff
@@ -2,27 +2,31 @@ if(typeof StatusHttpProvider === "undefined"){
 var callbackId = 0;
 var callbacks = {};
 
-function httpCallback(id, data) {
-    var result = data;
-    var error = null;
+WebViewBridge.onMessage = function (message) {
+    data = JSON.parse(message);
 
-    try {
-        result = JSON.parse(data);
-    } catch (e) {
-        error = {message: "InvalidResponse"};
-    }
+    if (data.type === "navigate-to-blank")
+        window.location.href = "about:blank";
 
-    if (callbacks[id]) {
-        callbacks[id](error, result);
+    else if (data.type === "status-api-success")
+    {
+        window.STATUS_API = data.data;
+        window.postMessage({ type: 'STATUS_API_SUCCESS', permissions: data.keys }, "*");
     }
-}
 
-var StatusHttpProvider = function (host, timeout) {
-    this.host = host || 'http://localhost:8545';
-    this.timeout = timeout || 0;
+    else if (data.type === "web3-send-async-callback")
+    {
+        var id = data.messageId;
+        if (callbacks[id]) {
+            callbacks[id](data.error, data.result);
+        }
+    }
 };
 
+var StatusHttpProvider = function () {};
+
 StatusHttpProvider.prototype.isStatus = true;
+StatusHttpProvider.prototype.isConnected = function () { return true; };
```

Comment:
Are those changes relevant?

Reply:
kind of, we had this function because we used http long time ago, i just simplified this function because node always 'connected' and moved it here